### PR TITLE
Ladder detector

### PR DESCRIPTION
**Pull request by sethtroisi**
This is a rough ladder detector. You can move all the printed files to a dir and hand inspect if you don't trust my metric.

Running over my cross-eval folder
```
20 (12  ): ***
21 (23  ): *******
22 (20  ): ******
23 (39  ): ************
24 (8   ): **
25 (24  ): *******
26 (6   ): *
27 (53  ): ****************
28 (7   ): **
29 (24  ): *******
30 (11  ): ***
31 (63  ): ********************
32 (5   ): *
33 (16  ): *****
34 (22  ): *******
35 (104 ): *********************************
36 (30  ): *********
37 (74  ): ***********************
38 (10  ): ***
39 (119 ): *************************************
40 (3   ): 
41 (6   ): *
42 (13  ): ****
43 (157 ): **************************************************
44 (1   ): 
45 (15  ): ****
47 (103 ): ********************************
49 (15  ): ****

0/93 (0.0%) /
0/120 (0.0%) 2018-07-25/
1/120 (0.8%) 2018-08-02/
7/319 (2.2%) 2018-09-02/
2/180 (1.1%) 2018-09-03/
0/119 (0.0%) 2018-09-04/
0/159 (0.0%) 2018-09-05/
0/233 (0.0%) 2018-09-06/
0/301 (0.0%) 2018-09-07/
0/121 (0.0%) 2018-09-08/
0/29 (0.0%) 2018-09-09/
3/984 (0.3%) 2018-09-14/
0/40 (0.0%) 2018-09-25/
0/30 (0.0%) 2018-09-26/
1/30 (3.3%) 2018-09-28/
8/840 (1.0%) 2018-10-01/
1/150 (0.7%) 2018-10-02/
1/313 (0.3%) 2018-10-03/
0/80 (0.0%) 2018-10-04/
1/200 (0.5%) 2018-10-05/
0/80 (0.0%) 2018-10-06/
18/1320 (1.4%) 2018-10-08/
0/81 (0.0%) 2018-10-09/
1/183 (0.5%) 2018-10-10/
0/50 (0.0%) 2018-10-11/
0/51 (0.0%) 2018-10-12/
0/50 (0.0%) 2018-10-15/
34/1355 (2.5%) 2018-10-24/
0/159 (0.0%) 2018-10-25/
1/252 (0.4%) 2018-10-26/
0/200 (0.0%) 2018-10-28/
1/471 (0.2%) 2018-10-29/
1/1747 (0.1%) 2018-10-30/
2/351 (0.6%) 2018-10-31/
0/655 (0.0%) 2018-11-01/
0/150 (0.0%) 2018-11-04/
0/500 (0.0%) 2018-11-08/
19/1624 (1.2%) 2018-11-09/
5/304 (1.6%) 2018-11-13/
2/100 (2.0%) 2018-12-12/
11/820 (1.3%) 2018-12-13/
1/80 (1.2%) 2018-12-23/
2/120 (1.7%) 2018-12-24/
3/243 (1.2%) 2018-12-25/
16/176 (9.1%) minigo-v-leela-10.games/
62/1440 (4.3%) minigo-v-leela-2.games/
43/1440 (3.0%) minigo-v-leela-3.games/
91/1440 (6.3%) minigo-v-leela-4.games/
124/1440 (8.6%) minigo-v-leela-5.games/
103/1440 (7.2%) minigo-v-leela-6.games/
55/1440 (3.8%) minigo-v-leela-7.games/
363/4320 (8.4%) minigo-v-leela.games/
983/28543 (3.4%) overall
```

### CODE REVIEW

File: oneoffs/ladder_detector.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import oneoff_utils
+
+
+ADJACENT_FOR_LADDER = 20
+
+
+def distance(a, b):
+    return (b[0] - a[0], b[1] - a[1])
+
+def absDistance(a, b):
```
Comment (tommadams) [inline oneoffs/ladder_detector.py]: s/absDistance/manhattanDistance/

File: oneoffs/ladder_detector.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+from tqdm import tqdm
+from sgfmill import sgf, sgf_moves
+
+import oneoff_utils
+
+
+ADJACENT_FOR_LADDER = 20
+
+
+def distance(a, b):
```
Comment (tommadams) [inline oneoffs/ladder_detector.py]: Distance is a scalar quantity. This should probably be called subtract or something

File: oneoffs/ladder_detector.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+def absDistance(a, b):
+    d = distance(a, b)
+    return abs(d[0]) + abs(d[1])
+
+def isLadderIsh(game_path):
+    with open(game_path) as sgf_file:
+        game_data = sgf_file.read().encode('utf-8')
+
+    g = sgf.Sgf_game.from_bytes(game_data)
+    board, moves = sgf_moves.get_setup_and_moves(g)
```
Comment (tommadams) [inline oneoffs/ladder_detector.py]: nit: s/board/_/ since it doesn't look like board is used

**Comment by sethtroisi** [comment]
After hand inspecting I found a bug and fixed it

39/176 (22.2%) minigo-v-leela-10.games/
155/1440 (10.8%) minigo-v-leela-2.games/
111/1440 (7.7%) minigo-v-leela-3.games/
275/1440 (19.1%) minigo-v-leela-4.games/
273/1440 (19.0%) minigo-v-leela-5.games/
226/1440 (15.7%) minigo-v-leela-6.games/
142/1440 (9.9%) minigo-v-leela-7.games/
851/4320 (19.7%) minigo-v-leela.games/
2072/13136 (15.8%) overall

**Pull request merged**